tiledb/sm/c_api/tiledb.h

```diff
@@ -3462,10 +3465,55 @@ TILEDB_EXPORT int32_t tiledb_query_get_config(
  *     layout has been set to `TILEDB_UNORDERED`. In the second case,
  *     if the user sets the layout to `TILEDB_UNORDERED` **after**
  *     the subarray has been set, the subarray will simply be ignored.
+ *
+ * @note This is a TILEDB_DEPRECATED api.
  */
 TILEDB_EXPORT int32_t tiledb_query_set_subarray(
     tiledb_ctx_t* ctx, tiledb_query_t* query, const void* subarray);
 
+/**
+ * Indicates that the query will write or read a subarray, and provides
+ * the appropriate information.
+ *
+ * **Example:**
+ *
+ * The following sets a 2D subarray [0,10], [20, 30] to the query.
+ *
+ * @code{.c}
+ * tiledb_subarray_t *subarray;
+ * tiledb_query_set_subarray(ctx, query, subarray);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param query The TileDB query.
+ * @param subarray The subarray by which the array read/write will be
+ *     constrained. It should be a sequence of [low, high] pairs (one
+ *     pair per dimension). For the case of writes, this is meaningful only
+ *     for dense arrays, and specifically dense writes. Note that `subarray`
+ *     must have the same type as the domain.
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ *
+ * @note If you set the subarray of a completed, or uninitialized
+ *     query, this function will clear the internal state and render it
+ *     as uninitialized. However, the potentially set layout and attribute
+ *     buffers will be retained. This is useful when the user wishes to
+ *     fix the attributes and layout, but explore different subarrays with
+ *     the same `tiledb_query_t` object (i.e., without having to create
+ *     a new object).
+ *
+ * @note Setting the subarray in sparse writes is meaningless and, thus,
+ *     this function will error in the following two cases, provided that
+ *     this is a write query:
+ *     (i) the array is sparse, and (ii) the array is dense and the
+ *     layout has been set to `TILEDB_UNORDERED`. In the second case,
+ *     if the user sets the layout to `TILEDB_UNORDERED` **after**
```

Comment:
Not yet changed
Consider prior precedents:
A) Item is direct copy/paste from elsewhere
B) Same highlighting used in several other (non (sub-)title-ish) locations (in addition to many are-(sub-)title-ish locations.)
C) Similar (possibly in error?, maybe just different markup?) with trailing single rather than double asterisk
```
**sometext*
```
also used in several locations.